library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/processor/RecordedDataProcessor.kt

```diff
@@ -165,37 +159,6 @@ internal class RecordedDataProcessor(
         }
     }
 
-    private fun executeRunnable(runnable: Runnable) {
-        @Suppress("SwallowedException", "TooGenericExceptionCaught")
-        try {
-            executorService.submit(runnable)
-        } catch (e: RejectedExecutionException) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-        } catch (e: NullPointerException) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // the task will never be null so normally this exception should not be triggered.
-            // In any case we will add a log here later.
-        }
-    }
-
-    private fun buildRunnable(
-        runnableFactory: (
-            timestamp: Long,
-            newContext: SessionReplayRumContext,
-            prevRumContext: SessionReplayRumContext
-        ) -> Runnable
-    ): Runnable? {
-        val rumContextData = rumContextDataHandler.createRumContextData() ?: return null
-
-        updateViewSent(rumContextData.newRumContext)
-
-        return runnableFactory(
-            rumContextData.timestamp,
-            rumContextData.newRumContext,
-            rumContextData.prevRumContext
-        )
-    }
-
     private fun updateViewSent(rumContext: SessionReplayRumContext) {
         // Because the runnable will be executed in another thread it can happen in case there is
```

Comment:
THis comment needs to be adjusted now -> `because this code is executed in the worker thread...`